app/src/main/java/dev/itsvic/parceltracker/api/DpdGerDeliveryService.kt

```diff
@@ -0,0 +1,90 @@
+package dev.itsvic.parceltracker.api
+
+import com.squareup.moshi.JsonClass
+import dev.itsvic.parceltracker.R
+import retrofit2.HttpException
+import retrofit2.Retrofit
+import retrofit2.http.GET
+import retrofit2.http.Path
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+object DpdGerDeliveryService : DeliveryService {
+    override val nameResource: Int
+        get() = R.string.service_dpd_ger
+    override val acceptsPostCode: Boolean
+        get() = false
+    override val requiresPostCode: Boolean
+        get() = false
+
+    override suspend fun getParcel(trackingId: String, postCode: String?): Parcel {
+        val resp = try {
+            service.getShipment(trackingId)
+        } catch (_: HttpException) {
+            throw ParcelNonExistentException()
+        }
+
+        val status = when(val current = resp.parcellifecycleResponse.parcelLifeCycleData.statusInfo.find {
+            it.isCurrentStatus == true
+        }?.status) {
+            "ACCEPTED" -> Status.Preadvice
+            "ON_THE_ROAD" -> Status.InTransit
+            "HANDOVER_CONSIGNOR_TO_PARCELSHOP", "AT_DELIVERY_DEPOT" -> Status.InWarehouse
+            "OUT_FOR_DELIVERY" -> Status.OutForDelivery
+            "DELIVERED" -> Status.Delivered
+            else -> logUnknownStatus("DPD (Europe)", current!!)
```

Comment:
```suggestion
            else -> logUnknownStatus("DPD Germany", current!!)
```
unless this will become a generalised implementation at one point, but until then, it should say germany specifically

Reply:
for some reason I thought first, that the german site could be used for packages in whole Europe, but that sadly is not the case